dimod/sampleset.py

```diff
@@ -1391,39 +1396,38 @@ def to_serializable(self, use_bytes=False, bytes_type=bytes):
         # bytes we could avoid a copy when undoing the serialization. However,
         # we want to pack the samples, so that means that we're storing the
         # arrays individually.
-        sample_data = {}
-        for name in self.record.dtype.names:
+        vectors = {name: serialize_ndarray(data, use_bytes=use_bytes,
+                                           bytes_type=bytes_type)
+                   for name, data in self.data_vectors.items()}
 
-            data = self.record[name]
-
-            if name == 'sample':
-                samples = self.record[name]
-                data = pack_samples(samples > 0)
-            else:
-                data = self.record[name]
-
-            if use_bytes:
-                data = bytes_type(data.tobytes(order='C'))
-            else:
-                data = self.record[name].tolist()
+        if self.vartype is Vartype.SPIN:
+            packed = pack_samples(self.record.sample > 0)
+        else:
+            packed = pack_samples(self.record.sample)
 
-            sample_data[name] = dict(data=data,
-                                     type=self.record[name].dtype.name,
-                                     shape=self.record[name].shape)
+        sample_data = serialize_ndarray(packed,
+                                        use_bytes=use_bytes,
+                                        bytes_type=bytes_type)
 
         return {
             # metadata
             "type": type(self).__name__,
             "version": {"sampleset_schema": schema_version},
-            "use_bytes": bool(use_bytes),
 
-            # sampleset
+            # samples
             "num_variables": len(self.variables),
             "num_rows": len(self),
+            "sample_data": sample_data,
+            "sample_type": self.record.sample.dtype.name,
+
+            # vectors
+            "vectors": vectors,
+
+            # other
             "variable_labels": self.variables.to_serializable(),
```

Comment:
Do we need to handle NumPy scalars/primitives for variables here (e.g. `int32`)?

Reply:
Good question, definitely an edge case we should test/handle.

Reply:
Handled by the Variables object https://github.com/dwavesystems/dimod/blob/368c8887948e62528a8369ea73744e2f584582d7/dimod/variables.py#L44